spec.emu

```diff
@@ -136,189 +175,307 @@ location: https://github.com/tc39/proposal-decimal/
+      <p>This method performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _O_ be the **this** value.
+        1. Perform ? RequireInternalSlot(_O_, [[Decimal128Data]]).
+        1. Perform ? RequireInternalSlot(_x_, [[Decimal128Data]]).
+        1. Let _d1_ be _O_.[[Decimal128Data]].
+        1. Let _d2_ be _x_.[[Decimal128Data]].
+        1. If _d1_ or _d2_ is *NaN*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+        1. If _d1_ is *+∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+        1. If _d1_ is *-∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+        1. Return Decimal128ValueToObject(« d1[0] * d2[0], min(d1[1], d2[1]) »).
+      </emu-alg>
     </emu-clause>
     <emu-clause id="sec-decimal.prototype.divide">
       <h1>Decimal128.prototype.divide ( _x_ )</h1>
-      <p>Divides *this* value by *_y_* according to the rules specified in the IEEE 754 Decimal128 specification. Neither *this* value nor *_x_* is modified. If, in the computation, more significant digits are generated than can be stored in an IEEE 754 Decimal128 value, the final digit will be computed by rounding according to <emu-xref href="#dfn-decimal128-default-rounding-mode">the default rounding mode</emu-xref>.</p>
+      <p>This method performs the following steps when called:</p>
+      <emu-alg>
+        1. Let _O_ be the **this** value.
+        1. Perform ? RequireInternalSlot(_O_, [[Decimal128Data]]).
+        1. Perform ? RequireInternalSlot(_x_, [[Decimal128Data]]).
+        1. Let _d1_ be _O_.[[Decimal128Data]].
+        1. Let _d2_ be _x_.[[Decimal128Data]].
+        1. If _d1_ or _d2_ is *NaN*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+        1. If _d1_ is *+∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*+∞*<sub>D</sub>).
+        1. If _d1_ is *-∞*<sub>D</sub>, then
+          1. If _d2_ is *+∞*<sub>D</sub>, return Decimal128ValueToObject(*NaN*<sub>D</sub>).
+          1. If _d2_ is *-∞*<sub>D</sub>, return Decimal128ValueToObject(*-∞*<sub>D</sub>).
+          1. Return Decimal128ValueToObject(*-∞*<sub>D</sub>).
```

Comment:
Also doubtful about the mathematics here

Reply:
Quite right. Fixed.